Sources/SnapshotPreviewsCore/UIKitRenderingStrategy.swift

```diff
@@ -33,38 +33,51 @@ public class UIKitRenderingStrategy: RenderingStrategy {
 
   @MainActor
   public func render(
-    preview: SnapshotPreviewsCore.Preview,
-    completion: @escaping (SnapshotResult) -> Void
+      preview: SnapshotPreviewsCore.Preview,
+      completion: @escaping (SnapshotResult) -> Void
   ) {
-    Self.setup()
-    let previewOrientation = preview.orientation.toInterfaceOrientation()
-    if windowScene!.interfaceOrientation != previewOrientation {
-      var rotationError: (any Error)? = nil
+      Self.setup()
+      let targetOrientation = preview.orientation.toInterfaceOrientation()
       if #available(iOS 16.0, *) {
-        windowScene!.requestGeometryUpdate(.iOS(interfaceOrientations: previewOrientation.toInterfaceOrientationMask())) { error in
-          NSLog("Rotation error handler: \(error) \(self.windowScene!.interfaceOrientation)")
-          rotationError = error
-        }
+          guard windowScene!.interfaceOrientation != targetOrientation else {
```

Comment:
What about adding on line 41:
```
guard #available(iOS 16.0, *), windowScene!.interfaceOrientation != targetOrientation else {
  performRender(preview: preview, completion: completion)
  return
}
```

Seems like that would remove one level of nesting and make it more readable